### DISCUSSION TITLE
Programatically detecting dropped frames

### DISCUSSION THREAD

**Discussion by naanlizard** [OvenMediaLabs/OvenMediaEngine#1518] [Help]
Hello everyone

We ingest arbitrary RTMP streams from our users, and they can be anywhere from perfect to completely garbage, will not load or generate frames. Presumably, there's a way to detect if a stream is broken (dropped/skipped/missed frames in OBS) using ffmpeg, but I'm not sure how. Anyone know how we might go about that?

**Comment by bchah** [comment]
Might not be exactly what you are looking for, but holy moly I just stumbled upon an impressive test suite Apple provides for analyzing HLS streams: https://developer.apple.com/documentation/http-live-streaming/using-apple-s-http-live-streaming-hls-tools

Besides this (which, based on a quick run just now will clearly give us an endless supply of red herrings to chase)... I wonder if the segment durations would change given dropped frames. FFmpeg could monitor and tell you that with something like:

`ffmpeg -I input_url.m3u8 -vf mpdecimate,setpts=N/FRAME_RATE/TB -f null -`
